src/Libraries/Microsoft.Extensions.AI/ChatCompletion/CachingChatClient.cs

```diff
@@ -3,10 +3,13 @@
 
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Shared.Diagnostics;
 
+#pragma warning disable S127 // "for" loop stop conditions should be invariant
```

Comment:
Just curious, why this wasn't needed before? Assume because before you were exclusively using foreach?

Reply:
The PR changed the core loop for how the coalescing is done. It now has a structure like:
```C#
for (int start = 0; start < length; start++)
{
    int end;
    for (end = start + 1; end < length; end++)
    {
        ...
    }
    ...

    start = end;
}
```
and that analyzer doesn't like updating the iteration variable in that way.

Reply:
Ah, understood. I was mainly curious since if we don't think it's valuable we could probably consider turning it off globally, but I can see how some bugs could be introduced given your example above.